azure.go

```diff
@@ -288,6 +332,12 @@ func (b *azureAuthBackend) getAzureSettings(ctx context.Context, config *azureCo
 		return nil, errors.New("resource is required")
 	}
 
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
```

Comment:
What is subscriptionID doing for rotate root? If I don't set this via the env or config, or if I set it to a bogus value I can still call rotate root and get a success. Maybe I am doing something wrong?

Reply:
@fairclothjm my understanding was that the subscription ID needs to be valid in order to set up a new client that allows us to talk to the MS Graph API. This happens on [this line](https://github.com/hashicorp/vault-plugin-auth-azure/pull/88/files#diff-f59e54a2e0d262d172de754ffdd049bf159deec9c5e0896fce921c2a0603b2dcR56) of our MS Graph implementation

When my credentials were invalid, I was having trouble with setting up the client and with talking to the API... not sure how a bogus value is yielding a valid client 🤔 @austingebauer do you have any thoughts on this?

Reply:
@fairclothjm, @austingebauer and I both did some digging into the implementation of the client in the `authorization` package, and it looks like the subscription ID is in fact not necessary to perform the rotate root operation, nice catch! Removed the extra bits of code that add the subscription ID. Since the sub ID is never being used, we are now simply passing an empty string into the `NewWithBaseURI` function [here](https://github.com/hashicorp/vault-plugin-auth-azure/pull/88/files#diff-6bf5bfdeab8247e592e67fdaf0494d289ff541b16cc55fad81542cbfdef15f1eR47)